v2/tools/generator/internal/armconversion/convert_from_arm_function_builder.go

```diff
@@ -174,78 +185,77 @@ func (builder *convertFromARMBuilder) namePropertyHandler(
 			"ExtractKubernetesResourceNameFromARMName",
 			astbuilder.Selector(dst.NewIdent(builder.typedInputIdent), string(fromProp.PropertyName()))))
 
-	return astbuilder.Statements(
-		setAzureName), true
+	return handleWith(setAzureName), nil
 }
 
 func (builder *convertFromARMBuilder) userAssignedIdentitiesPropertyHandler(
 	toProp *astmodel.PropertyDefinition,
 	_ *astmodel.ObjectType,
-) ([]dst.Stmt, bool) {
+) (propertyConversionHandlerResult, error) {
 	if _, ok := astmodel.IsUserAssignedIdentityProperty(toProp); !ok {
-		return nil, false
+		return notHandled, nil
 	}
 
 	// TODO: For now we are not assigning these, as we don't know how to rebuild the reference
-	return nil, true
+	return handledWithNOP, nil
 }
 
 func (builder *convertFromARMBuilder) referencePropertyHandler(
 	toProp *astmodel.PropertyDefinition,
 	_ *astmodel.ObjectType,
-) ([]dst.Stmt, bool) {
+) (propertyConversionHandlerResult, error) {
 	if !astmodel.IsTypeResourceReference(toProp.PropertyType()) {
-		return nil, false
+		return notHandled, nil
 	}
 
 	// TODO: For now, we are NOT assigning to these. _Status types don't have them and it's unclear what
 	// TODO: the fromARM functions do for us on Spec types. We may need them for diffing though. If so we will
 	// TODO: need to revisit this and actually assign something
-	return nil, true
+	return handledWithNOP, nil
 }
 
 func (builder *convertFromARMBuilder) secretPropertyHandler(
 	toProp *astmodel.PropertyDefinition,
 	_ *astmodel.ObjectType,
-) ([]dst.Stmt, bool) {
+) (propertyConversionHandlerResult, error) {
 	isSecretReference := astmodel.TypeEquals(toProp.PropertyType(), astmodel.SecretReferenceType)
 	isOptionalSecretReference := astmodel.TypeEquals(toProp.PropertyType(), astmodel.OptionalSecretReferenceType)
 
 	if !isSecretReference && !isOptionalSecretReference {
-		return nil, false
+		return notHandled, nil
 	}
 
 	// TODO: For now, we are NOT assigning to these. _Status types don't have them and it's unclear what
 	// TODO: the fromARM functions do for us on Spec types. We may need them for diffing though. If so we will
 	// TODO: need to revisit this and actually assign something
-	return nil, true
+	return handledWithNOP, nil
```

Comment:
is `NOP` supposed to be `no-op`? Shouldn't that be `handledWithNoOp`?

Reply:
Yes, `NOP` == no-operation. I guess that abbreviation is hard coded in my brain from when I learned 6502 and Z80 assembly programming back in High School. 😁 
I'll rename it as suggested.

Reply:
Done.